soroban-sdk/src/token.rs

```diff
@@ -139,9 +139,17 @@ pub trait TokenInterface {
     ///
     /// # Events
     ///
-    /// Emits an event with topics `["transfer", from: Address, to: Address],
-    /// data = amount: i128`
-    fn transfer(env: Env, from: Address, to: Address, amount: i128);
+    /// Emits an event with topics
+    /// `["transfer", from: Address, to: Address]
+    /// If `to` is not a muxed address (i.e. has no muxed_id set), then data
+    /// will  have the following format (the `Transfer` event defined below):
+    /// `data = amount: i128`
+    ///
+    /// If `to` is a muxed address (i.e. has a muxed_id set), then data will
+    /// have be emitted as map with Symbol keys in the following format (the
+    /// `TransferMuxed` event defined below):
+    /// `data = { "amount": i128, "to_muxed_id": u64 }`
```

Comment:
Fyi I've opened a couple issues to improve how we communicate about the shape of the published events:
- #1519 
- #1520

Reply:
Yeah, we can update this then.